src/Orleans/Configuration/ClientConfiguration.cs

```diff
@@ -137,6 +138,11 @@ public enum GatewayProviderType
         public bool StatisticsWriteLogStatisticsToTable { get; set; }
         public StatisticsLevel StatisticsCollectionLevel { get; set; }
 
+        /// <summary>
+        /// Gets or sets the service configuration delegate.
+        /// </summary>
+        public Func<IServiceCollection, IServiceProvider> ServiceConfigurator { get; set; } = services => services.BuildServiceProvider();
```

Comment:
In a general sense this is functionality not configuration.  Does this belong here?